### PR TITLE
perf(group_owners): Improve performance of `get_previous_releases` for orgs with a large number of releases

### PR DESCRIPTION
**Pull request by wedamija**
Similar to work done in https://github.com/getsentry/sentry/pull/11814.

For orgs with a large number of releases, trying to fetch releases for a project ordered by release
date ends up being slow due to the way the tables end up joining. To fix this, we hack in a subquery
that limits how many rows are considered, which makes the query substantially faster. This works
because `Release.id` is closely correlated to release date, and provides a reasonable proxy. It will
likely be wrong in some edge cases, but is accurate enough for our uses in suspect commits.

We should likely denormalize the coalesce of `date_released` and `date_added` onto `ReleaseProject`
so that we can make these queries better.

### CODE REVIEW

File: src/sentry/utils/committers.py [hunk 1]
```diff
@@ -156,26 +156,43 @@ def get_previous_releases(project, start_version, limit=5):
...
+                    """
+                        SELECT sr.*
+                        FROM sentry_release as sr
+                        INNER JOIN (
+                            SELECT release_id
+                            FROM sentry_release_project
+                            WHERE project_id = %s
+                            AND sentry_release_project.release_id <= %s
+                            ORDER BY release_id desc
+                            LIMIT 100
```
Comment (wedamija) [inline src/sentry/utils/committers.py:187]: I chose this arbitrarily, we could even lower the number if we want this to be faster, but should be fast enough as is.

**Review by iProgramStuff — APPROVED** [review]

File: src/sentry/utils/committers.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/utils/committers.py:185]: Maybe a good idea to make this a configured parameter?

Comment (wedamija) [reply src/sentry/utils/committers.py:185]: It'll require a code change either way. If this becomes something that we do need to configure I'll add it in later, but based on https://github.com/getsentry/sentry/pull/11814 we will probably never touch this again (unless we do the denormalization)

Comment (EvanPurkhiser) [reply src/sentry/utils/committers.py:185]: 👍

Comment (dcramer) [reply src/sentry/utils/committers.py:185]: only feedback id give (from authoring this kind of thing a lot) is up the 100 to something more significant unless theres a serious cost

e.g. 1k or 10k

**Pull request merged**